File: cs/CorreosCS/CorreosCS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace CorreosCS
{

	









	public partial class Form1 : Form
	{

		public bool arroba;

		public Form1()
		{
			InitializeComponent();
		}












		private void btnValidar_Click(object sender, EventArgs e)
		{

			if ((LikeOperator.LikeString(txtCorreo.Text, "[a-z]*@[!.]?*.[a-z][a-z][a-z].[a-z][a-z]", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "[a-z]*@[!.]?*.[a-z][a-z][a-z]", CompareMethod.Binary)) & !LikeOperator.LikeString(txtCorreo.Text, "*@*@*", CompareMethod.Binary) & !LikeOperator.LikeString(txtCorreo.Text, "*@*.*.*.*", CompareMethod.Binary) & !LikeOperator.LikeString(txtCorreo.Text, "*--*", CompareMethod.Binary) & !LikeOperator.LikeString(txtCorreo.Text, "*..*", CompareMethod.Binary))
			{
				if (LikeOperator.LikeString(txtCorreo.Text, "*[/]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[!]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[*]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[,]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[[]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[]]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, @"*[\]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[+]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[=]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[{]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[}]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[|]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[&]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[(]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[)]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[;]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[:]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[']*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[^]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[%]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[$]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[#]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[`]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[~]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[<]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[>]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[ ]*", CompareMethod.Binary) | LikeOperator.LikeString(txtCorreo.Text, "*[?]*", CompareMethod.Binary))
				{
					MessageBox.Show("Correo Invalido");
				}
				else
				{
					MessageBox.Show("Correo Valido");
				}
			}
			else
			{
				MessageBox.Show("Correo Invalido");
			}

		}





	}
}

File: vb/CorreoVB/CorreoVB/Form1.vb
Public Class frmCorreoVB

    Dim arroba As Boolean

    Private Sub btnValidar_Click(sender As Object, e As EventArgs) Handles btnValidar.Click

        If (((txtCorreo.Text Like "[a-z]*@[!.]?*.[a-z][a-z][a-z].[a-z][a-z]") Or (txtCorreo.Text Like "[a-z]*@[!.]?*.[a-z][a-z][a-z]")) And Not (txtCorreo.Text Like "*@*@*") And Not (txtCorreo.Text Like "*@*.*.*.*") And Not (txtCorreo.Text Like "*--*") And Not (txtCorreo.Text Like "*..*")) Then

            If ((txtCorreo.Text Like "*[/]*") Or (txtCorreo.Text Like "*[!]*") Or (txtCorreo.Text Like "*[*]*") Or (txtCorreo.Text Like "*[,]*") Or (txtCorreo.Text Like "*[[]*") Or (txtCorreo.Text Like "*[]]*") Or (txtCorreo.Text Like "*[\]*") Or (txtCorreo.Text Like "*[+]*") Or (txtCorreo.Text Like "*[=]*") Or (txtCorreo.Text Like "*[{]*") Or (txtCorreo.Text Like "*[}]*") Or (txtCorreo.Text Like "*[|]*") Or (txtCorreo.Text Like "*[&]*") Or (txtCorreo.Text Like "*[(]*") Or (txtCorreo.Text Like "*[)]*") Or (txtCorreo.Text Like "*[;]*") Or (txtCorreo.Text Like "*[:]*") Or (txtCorreo.Text Like "*[']*") Or (txtCorreo.Text Like "*[^]*") Or (txtCorreo.Text Like "*[%]*") Or (txtCorreo.Text Like "*[$]*") Or (txtCorreo.Text Like "*[#]*") Or (txtCorreo.Text Like "*[`]*") Or (txtCorreo.Text Like "*[~]*") Or (txtCorreo.Text Like "*[<]*") Or (txtCorreo.Text Like "*[>]*") Or (txtCorreo.Text Like "*[ ]*") Or (txtCorreo.Text Like "*[?]*")) Then

                MsgBox("Correo Invalido")

            Else

                MsgBox("Correo Valido")

            End If



        Else

            MsgBox("Correo Invalido")

        End If


    End Sub



End Class
